### ISSUE TITLE
Rule proposal: consistent built-in iterator access style

### ISSUE THREAD

**Issue by jonahsnider** [sindresorhus/eslint-plugin-unicorn#1845]
### Description

Many built-in data structures (ex. `Map`) expose an iterator with the well known `@@iterator` (`Symbol.iterator`) key and the `#entries()` instance method.

Some built-in data structures (ex. `Array`) are the same but with `@@iterator` and `#values()`.

Some built-in data structures (ex. `Set`) are the same but with `@@iterator` and `#keys()`.

Some built-in data structures support all 3 methods for the same iterator. I think `Set` is the only one that does this.

Having a rule to ensure consistent usage of these iterators can be helpful for maintaining convention in a codebase.

### Fail

When preferring `@@iterator`:
```js
for (const x of map.entries()) {}
```
```js
[...map.entries()];
```
```js
new Map(map.entries());
```
```js
[...array.values()];
```

When preferring `#entries()` or `#values()`:
```js
for (const x of map) {}
```
```js
[...map];
```
```js
new Map(map);
```
```js
[...array];
```

### Pass

When preferring `#entries()`:
```js
for (const x of map.entries()) {}
```
```js
[...map.entries()];
```
```js
new Map(map.entries());
```

When preferring `@@iterator`:
```js
for (const x of map) {}
```
```js
[...map];
```
```js
new Map(map);
```

### Additional Info

Built-in data structures that should support `@@iterator` and `#entries()`:
- `Map`
- `Set`
- `FormData` (the global, not the one from `form-data`)

Built-in data structures that should support `@@iterator` and `#values()`:
- `Array`
- `Set`
- `TypedArray`

Built-in data structures that should support `@@iterator` and `#keys()`:
- `Set`

**Comment by sindresorhus** [comment]
I see absolutely no reason to prefer `.entries()` over `@@iterator`. `.entries()` only exists because of legacy reasons. So I think the rule should always enforce `@@iterator`.

However, I do see the case for whether to prefer `for (const [, value] of foo) {}` or `for (const value of foo.values()) {}`.
